### PR TITLE
Add n8k support

### PR DESCRIPTION
**Pull request by saichint**
For n8k, the CLIs for port-channel global are different. This is for adding support to n8k. Also the default value for lacp_max_bundle is 32 for n8k.

All tests passed for all switches except n3k where this provider does not work unless it is put in n9k mode.

Test results on n8k:
[saichint@sjc-ads-7106 /ws/saichint-sjc/puppet_agent/cisco-network-node-utils/tests] $ ruby test_portchannel_global.rb -e n8k -v  
Run options: -e n8k -v --seed 41452                                                                                 
# Running:

Node under test:
- name  - n9k-157
- type  - N85-C8508
- image - bootflash:///n8500-dk9.7.0.3.F1.0.267.bin

TestPortchannelGlobal#test_load_balance_hash_poly = 2.85 s = S
TestPortchannelGlobal#test_load_balance_no_hash_rot = 5.17 s = .
TestPortchannelGlobal#test_hash_distribution = 2.64 s = .  
TestPortchannelGlobal#test_load_defer = 2.68 s = .  
TestPortchannelGlobal#test_load_balance_sym_concat_rot = 2.75 s = S
TestPortchannelGlobal#test_load_balance_asym_rot = 2.69 s = S  
TestPortchannelGlobal#test_resilient = 2.70 s = .                  

Finished in 23.916315s, 0.2927 runs/s, 0.5017 assertions/s.

  1) Skipped:
TestPortchannelGlobal#test_load_balance_hash_poly [test_portchannel_global.rb:169]:
Test not supported on this platform                                                

  2) Skipped:
TestPortchannelGlobal#test_load_balance_sym_concat_rot [test_portchannel_global.rb:113]:
Test not supported on this platform                                                     

  3) Skipped:
TestPortchannelGlobal#test_load_balance_asym_rot [test_portchannel_global.rb:197]:
Test not supported on this platform                                               

7 runs, 12 assertions, 0 failures, 0 errors, 3 skips
Coverage report generated for MiniTest to /ws/saichint-sjc/puppet_agent/cisco-network-node-utils/tests/coverage. 0.0 / 0.0 LOC (100.0%) covered.

### CODE REVIEW

File: lib/cisco_node_utils/cmd_ref/interface_portchannel.yaml [hunk 1]
```diff
@@ -28,6 +28,8 @@ lacp_max_bundle:
   set_value: "lacp max-bundle %s"
   N3k:
```
Comment (mikewiebe) [inline lib/cisco_node_utils/cmd_ref/interface_portchannel.yaml]: Better to start using yaml anchor and aliases here.

https://github.com/cisco/cisco-network-node-utils/blob/develop/lib/cisco_node_utils/cmd_ref/README_YAML.md#yaml-anchors-and-aliases

File: tests/test_portchannel_global.rb [hunk 2]
```diff
@@ -215,4 +215,26 @@ def test_load_balance_asym_rot
     assert_equal(global.default_asymmetric, global.asymmetric)
     assert_equal(global.default_rotate, global.rotate)
   end
+
+  def test_load_balance_no_hash_rot
+    global = create_portchannel_global
+    unless validate_property_excluded?('portchannel_global', 'hash_poly')
```
Comment (mikewiebe) [inline tests/test_portchannel_global.rb]: This does not look right.  Shouldn't this be `if validate_property_excluded?`

Comment (mikewiebe) [inline tests/test_portchannel_global.rb]: Also, I would prefer you test for unsupported and nil instead of skipping if possible.

Example:

https://github.com/saichint/cisco-network-node-utils/blob/test/pc/tests/test_portchannel_global.rb#L56

Comment (saichint) [inline tests/test_portchannel_global.rb]: Let us discuss this. The code is correct even though it looks odd, I had to do this way as I could not find a cleaner approach
